app/controllers/rentals_controller.rb

```diff
@@ -22,6 +23,27 @@ def index
   def show
   end
 
+  # GET /rentals/search_users?q
+  def search_users
+    # prioritiezed by order queries
+    query_priority = %i(email_cont spire_id_eq full_name_cont)
+    @users = []
+
+    query_priority.each do |q|
+      @users += User.ransack(q => params[:user_search_query]).result
+      # break if @users.present? # maybe just join all the lists
+    end
+
+    # could be a department too
+    ## join users with departments
+    @users += Department.ransack(name_cont: params[:user_search_query]).result.map(&:users).flatten
+
+    @users = User.all if @users.blank? # if we couldnt find anything pass them everything
+
+    @users = @users.uniq.paginate(page: params[:page], per_page: 8) # remove duplicates and split into pages
```

Comment:
Like on line 33?
It needs to be done in one centralize location afterwards because it will search by all types.
If i have a User(first_name: "jim", email: "jim@gmail.com"), I will match the same user twice.